kaolin/datasets/modelnet.py

```diff
@@ -47,143 +41,129 @@ class ModelNet10(torch.utils.data.Dataset):
-            return pts, label
-        else:
-            raise NotImplementedError
-
-
-class ModelNetVoxels(object):
+        return len(self.names)
+
+    def __getitem__(self, index):
+        """Returns the item at index idx. """
+        category = torch.tensor(self.cat_idxs[index], dtype=torch.long, device=self.device)
+        data = TriangleMesh.from_off(self.filepaths[index])
+        data.to(self.device)
+        if self.transform:
+            data = self.transform(data)
+
+        return data, category
+
+
+class ModelNet_Voxels(object):
     r""" Dataloader for downloading and reading from ModelNet.
 
 
     Args:
-        root (str): location the dataset should be downloaded to /loaded from
+        basedir (str): location the dataset should be downloaded to /loaded from
         train (bool): if True loads training set, else loads test
-        download (bool): downloads the dataset if not found in root
+        download (bool): downloads the dataset if not found in basedir
         categories (str): list of object classes to be loaded
-        single_view (bool): if true only one roation is used, if not all 12 views are loaded
 
     Returns:
         .. code-block::
-        
+
         dict: {
             'attributes': {'name': str, 'class': str},
             'data': {'voxels': torch.Tensor}
         }
 
 
     Examples:
-        >>> dataset = ModelNet(root='../data/')
+        >>> dataset = ModelNet(basedir='../data/')
         >>> train_loader = DataLoader(dataset, batch_size=10, shuffle=True, num_workers=8)
         >>> obj = next(iter(train_loader))
         >>> obj['data']['data']
         torch.Size([10, 30, 30, 30])
     """
 
-    def __init__(self, root: str = '../data/', train: bool = True, test: bool = True,
-                 download: bool = True, categories: list = ['chair'], single_view: bool = True):
-        if not os.path.exists(root + '/ModelNet/'):
-            assert download, "ModelNet is not found, and download is set to False"
-            assert (train or test), 'either train or test must be set to True'
-
-        if not single_view:
-            side = ''
-        else:
-            side = '_1'
-
-        all_classes = [os.path.basename(os.path.dirname(c))
-                       for c in glob(root + '/ModelNet/volumetric_data/*/')]
-        self.names = []
-        for category in categories:
-            assert category in all_classes, 'object class {0} not in list of availible classes: {1}'.format(
-                category, all_classes)
-            if train:
-                self.names += glob(
-                    root + '/ModelNet/volumetric_data/{0}/*/{1}/*{2}.mat'.format(category, 'train', side))
-            if test:
-                self.names += glob(
-                    root + '/ModelNet/volumetric_data/{0}/*/{1}/*{2}.mat'.format(category, 'test', side))
+    def __init__(self, basedir: str = 'data/ModelNet/', cache_dir: str = 'cache', train: bool = True, categories: list = ['bed'],
```

Comment:
1) I think `basedir` should be a required argument with no default value.
2) `basedir` don't have underscore but `cache_dir` does.

Reply:
Agreed this is not ideal. `cache_dir` follows the format of the ShapeNet datasets, while `basedir` is the argument name used by the examples. This will need to be corrected in the future to ensure consistency between all dataset classes.